addons/sale/controllers/portal.py

```diff
@@ -175,18 +175,19 @@ def _get_payment_values(self, order_sudo):
         :return: The payment-specific values.
         :rtype: dict
         """
-        logged_in = not request.env.user._is_public()
         providers_sudo = request.env['payment.provider'].sudo()._get_compatible_providers(
             order_sudo.company_id.id,
             order_sudo.partner_id.id,
             order_sudo.amount_total,
             currency_id=order_sudo.currency_id.id,
             sale_order_id=order_sudo.id,
         )  # In sudo mode to read the fields of providers and partner (if not logged in)
-        tokens = request.env['payment.token'].search([
-            ('provider_id', 'in', providers_sudo.ids),
-            ('partner_id', '=', order_sudo.partner_id.id)
-        ]) if logged_in else request.env['payment.token']
+
+        # Even in the case of the partner not logged in, let the tokens accessible to the partner
+        # to be used.
```

Comment:
I don't think that this comment is useful but if you want to keep it, please use it everywhere (in `account_payment` for instance)

Reply:
you're right, the code is self-explanatory :)